src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -0,0 +1,1845 @@
+                //         FA(() => 0);
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "() => 0").WithArguments("inferred delegate type").WithLocation(18, 12),
+                // (19,12): error CS8652: The feature 'inferred delegate type' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         FB(() => { });
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "() => { }").WithArguments("inferred delegate type").WithLocation(19, 12),
+                // (22,12): error CS8652: The feature 'inferred delegate type' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         FA(delegate () { return 0; });
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "delegate () { return 0; }").WithArguments("inferred delegate type").WithLocation(22, 12),
+                // (23,12): error CS8652: The feature 'inferred delegate type' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         FB(delegate () { });
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "delegate () { }").WithArguments("inferred delegate type").WithLocation(23, 12));
+
+            CompileAndVerify(source, parseOptions: TestOptions.RegularPreview, expectedOutput:
+@"FA(Action)
+FA(Delegate)
+FB(Delegate)
+FB(Func<int>)
+FA(Action)
+FA(Delegate)
+FB(Delegate)
+FB(Func<int>)
+FA(Action)
+FA(Delegate)
+FB(Delegate)
+FB(Func<int>)
+");
+        }
+
+        [Fact]
+        public void OverloadResolution_04()
+        {
+            var source =
+@"using System;
+class Program
+{
+    static void Main()
+    {
+        var c = new C();
+        c.M(Main);
+        c.M(() => { });
+    }
+}
+class C
+{
+    public void M(Delegate d) { Console.WriteLine(""C.M""); }
+}
+static class E
+{
+    public static void M(this object o, Action a) { Console.WriteLine(""E.M""); }
+}";
+
+            var comp = CreateCompilation(source, parseOptions: TestOptions.Regular9);
+            comp.VerifyDiagnostics(
+                // (7,13): error CS8652: The feature 'inferred delegate type' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         c.M(Main);
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "Main").WithArguments("inferred delegate type").WithLocation(7, 13),
+                // (8,13): error CS8652: The feature 'inferred delegate type' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         c.M(() => { });
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "() => { }").WithArguments("inferred delegate type").WithLocation(8, 13));
+
+            CompileAndVerify(source, parseOptions: TestOptions.RegularPreview, expectedOutput:
+@"C.M
```

Comment:
It would be good to comment on behavior change. Should document the break in compiler doc.

Reply:
The behavior has not changed yet.

Reply:
Copy&pasting this test into [sharplab](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUBgAlUwHY8BvPA6ozANiIBYCBZAQwEsoAKASipqVcNEQQBubMAQDGBALwEoAUwDuBAMJ8A3ANEyAdC27suvHcL0GjfeQD4C5AgF8zu53id5C6im9QBmJlZuABElABslAHM2YCUCOF4HWgBObgAidUN0s3dcT3x6InQCAFFfC2oA2gZUZiNgAAsOAGcCAHsAIwArJWlgDphaErYkx2I09NLs3IKnIA===) I get `E.M" in output.

---
In reply to: [618955062](https://github.com/dotnet/roslyn/pull/52448#discussion_r618955062) [](ancestors = 618955062)

Reply:
You're right, thanks. I've added a comment and updated the breaking change document.